Sources/SwiftDocC/Model/Rendering/RenderNodeTranslator.swift

```diff
@@ -74,6 +76,21 @@ public struct RenderNodeTranslator: SemanticVisitor {
         return assetReference
     }
     
+    private func fileContents(with fileReference: ResourceReference) -> String? {
+        // Check if the file is a local asset that can be read directly from the context
+        if let fileData = try? context.resource(with: fileReference) {
```

Comment:
Nit: It looks like this logic is duplicated at least once – should we consider moving this to a specific method in the `DocumentationContext`?

Reply:
Twice is below my threshold for duplication of something this small. I'm also thinking that this behavior would probably conceptually be more relevant to the asset manager or the data provider but moving it to either of those places would be bigger than this.